openrouteservice/src/main/java/org/heigit/ors/isochrones/builders/concaveballs/ConcaveBallsIsochroneMapBuilder.java

```diff
@@ -332,6 +332,7 @@ private GeometryCollection buildIsochrone(AccessibilityMap edgeMap, List<Coordin
 		GraphHopperStorage graph = searchContext.getGraphHopper().getGraphHopperStorage();
 		NodeAccess nodeAccess = graph.getNodeAccess();
 		int maxNodeId = graph.getNodes();
```

Comment:
In node indexing is 0-based and `getNodes()` returns the number of nodes, then shouldn't this rather be `int maxNodeId = graph.getNodes() - 1;`?

Reply:
Apparently that's not entirely true according to https://github.com/GIScience/graphhopper/blob/ors_0.13/core/src/main/java/com/graphhopper/routing/QueryGraph.java#L375

Reply:
Yes, as I said, for the edges it is not true (I use id > maxId -1, QueryGraph uses id >= maxId). So seems that its also not true for the nodes, if we can trust QueryGraph impl. So generally the GH explanation is wrong.

Reply:
Changed it

Reply:
I believe both `isVirtualEdge` and `isVirtualNode` methods are correct considering other bits of GH code I've seen such as https://github.com/GIScience/graphhopper/blob/ors_0.13/core/src/main/java/com/graphhopper/storage/BaseGraph.java#L69-L72 and https://github.com/GIScience/graphhopper/blob/ors_0.13/core/src/main/java/com/graphhopper/storage/BaseGraph.java#L86-89, respectively. How